controllers/drplacementcontrol_controller.go

```diff
@@ -1099,18 +1119,29 @@ func getDRClusters(ctx context.Context, client client.Client, drPolicy *rmn.DRPo
 }
 
 func (r *DRPlacementControlReconciler) processDeletion(ctx context.Context,
-	drpc *rmn.DRPlacementControl, placementObj client.Object, log logr.Logger,
+	drpc *rmn.DRPlacementControl, log logr.Logger,
 ) error {
 	log.Info("Processing DRPC deletion")
 
 	if !controllerutil.ContainsFinalizer(drpc, DRPCFinalizer) {
 		return nil
 	}
 
+	placementObj, err := getPlacementOrPlacementRule(ctx, r.Client, drpc, log)
+	if err != nil && !errors.IsNotFound(err) {
+		// It is ok to not find the placementObj if the drpc is being deleted
+		return fmt.Errorf("failed to get User PlacementRule/Placement %w", err)
+	}
+
```

Comment:
@raghavendra-talur, in case that error is "not found", are we proceeding? Then we'll fail [here](https://github.com/RamenDR/ramen/blob/930463d87fcbf91eda11dc43175351a570298dc2/controllers/drplacementcontrol_controller.go#L2103)